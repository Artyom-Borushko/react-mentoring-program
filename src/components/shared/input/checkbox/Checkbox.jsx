import React from 'react';
import { useField } from 'formik';
import '../input.css';

export default function Checkbox({ ...props }) {
  const [field] = useField({ ...props, type: 'checkbox' });

  return (
    <div className="input-wrapper">
      <label className="checkbox-input" htmlFor={field.id}>
        <input type="checkbox" {...field} {...props} />
        {props.label}
      </label>
      {/* Handle validation outside of this component
      to not display an error message next to each checkbox option */}
    </div>
  );
}
