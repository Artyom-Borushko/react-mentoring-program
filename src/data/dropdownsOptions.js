export const movieDropdownOptions = ['edit', 'delete'];

export const moviesSortDropdownOptions = ['release date', 'title'];
