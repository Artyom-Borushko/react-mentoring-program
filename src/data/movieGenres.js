
export const listOfGenres = ['all', 'documentary', 'comedy', 'horror', 'crime'];
