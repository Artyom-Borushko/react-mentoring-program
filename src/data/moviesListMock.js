export const listOfMoviesMock = [
    { src: '/images/movies-list/pulpFiction.png', title: "Pulp Fiction", releaseYear: "2004", relevantGenres: ['action', 'crime'], rating: '9.2', duration: '2h 22min', description: 'Lorem ipsum dolor sit amet. Eum autem quod aut dolorem voluptatum nam dolores sunt. Ea velit placeat ea corporis illo rem quod nulla. Et maxime placeat ut vero placeat et harum sequi. A Quis expedita id dolores sint id repellendus atque sit suscipit obcaecati vel quia galisum quo cumque internos!', id: "uniqueMovieId1" },
    { src: '/images/movies-list/bohemianRhapsody.png', title: "Bohemian Rhapsody", releaseYear: "2003", relevantGenres: ['documentary', 'drama'], rating: '8.4', duration: '1h 27min', description: 'Lorem ipsum dolor sit amet. Eum autem quod aut dolorem voluptatum nam dolores sunt. Ea velit placeat ea corporis illo rem quod nulla. Et maxime placeat ut vero placeat et harum sequi. A Quis expedita id dolores sint id repellendus atque sit suscipit obcaecati vel quia galisum quo cumque internos!', id: "uniqueMovieId2" },
    { src: '/images/movies-list/killBillVol2.png', title: "Kill Bill: Vol 2", releaseYear: "1994", relevantGenres: ['comedy', 'crime', 'oscar'], rating: '6.2', duration: '3h 12min', description: 'Lorem ipsum dolor sit amet. Eum autem quod aut dolorem voluptatum nam dolores sunt. Ea velit placeat ea corporis illo rem quod nulla. Et maxime placeat ut vero placeat et harum sequi. A Quis expedita id dolores sint id repellendus atque sit suscipit obcaecati vel quia galisum quo cumque internos!', id: "uniqueMovieId3" },
    { src: '/images/movies-list/avengersWarOfInfinity.png', title: "Avengers: War of Infinity", releaseYear: "2004", relevantGenres: ['action', 'adventure'], rating: '8.2', duration: '0h 56min', description: 'Lorem ipsum dolor sit amet. Eum autem quod aut dolorem voluptatum nam dolores sunt. Ea velit placeat ea corporis illo rem quod nulla. Et maxime placeat ut vero placeat et harum sequi. A Quis expedita id dolores sint id repellendus atque sit suscipit obcaecati vel quia galisum quo cumque internos!', id: "uniqueMovieId4" },
    { src: '/images/movies-list/inception.png', title: "Inception", releaseYear: "2003", relevantGenres: ['action', 'adventure', 'horror'], rating: '10.0', duration: '1h 39min', description: 'Lorem ipsum dolor sit amet. Eum autem quod aut dolorem voluptatum nam dolores sunt. Ea velit placeat ea corporis illo rem quod nulla. Et maxime placeat ut vero placeat et harum sequi. A Quis expedita id dolores sint id repellendus atque sit suscipit obcaecati vel quia galisum quo cumque internos!', id: "uniqueMovieId5" },
    { src: '/images/movies-list/reservoirDogs.png', title: "Reservoir dogs", releaseYear: "1994", relevantGenres: ['crime', 'oscar'], rating: '7.7', duration: '2h 00min', description: 'Lorem ipsum dolor sit amet. Eum autem quod aut dolorem voluptatum nam dolores sunt. Ea velit placeat ea corporis illo rem quod nulla. Et maxime placeat ut vero placeat et harum sequi. A Quis expedita id dolores sint id repellendus atque sit suscipit obcaecati vel quia galisum quo cumque internos!', id: "uniqueMovieId6" },
];
