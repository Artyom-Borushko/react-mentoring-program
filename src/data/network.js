export default 'http://localhost:4000/movies';
