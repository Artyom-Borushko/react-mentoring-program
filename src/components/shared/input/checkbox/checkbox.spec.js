import {render} from "@testing-library/react";
import React from "react";
import '@testing-library/jest-dom';
import {Formik} from "formik";
import Checkbox from "./Checkbox";
import {movieGenres} from "../../../../data/movieGenres";

describe("Checkbox input component tests", () => {

    it("renders checkbox input with Formik", () => {
        const { container } = render(
            <Formik
                initialValues={{ movieOverview: "" }}
                onSubmit={() => {}}
            >
                <div>
                    {movieGenres.map((genre) => (
                        <Checkbox
                            key={genre}
                            name="movieGenres"
                            id="movieGenres"
                            value={genre}
                            label={genre}
                        />
                    ))}
                </div>
            </Formik>
        );

        expect(container).toMatchSnapshot();
    });
});
