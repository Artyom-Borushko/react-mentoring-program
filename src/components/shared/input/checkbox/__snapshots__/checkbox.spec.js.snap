// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox input component tests renders checkbox input with Formik 1`] = `
<div>
  <div>
    <div
      class="input-wrapper"
    >
      <label
        class="checkbox-input"
      >
        <input
          id="movieGenres"
          label="Animation"
          name="movieGenres"
          type="checkbox"
          value="Animation"
        />
        Animation
      </label>
    </div>
    <div
      class="input-wrapper"
    >
      <label
        class="checkbox-input"
      >
        <input
          id="movieGenres"
          label="Adventure"
          name="movieGenres"
          type="checkbox"
          value="Adventure"
        />
        Adventure
      </label>
    </div>
    <div
      class="input-wrapper"
    >
      <label
        class="checkbox-input"
      >
        <input
          id="movieGenres"
          label="Family"
          name="movieGenres"
          type="checkbox"
          value="Family"
        />
        Family
      </label>
    </div>
    <div
      class="input-wrapper"
    >
      <label
        class="checkbox-input"
      >
        <input
          id="movieGenres"
          label="Comedy"
          name="movieGenres"
          type="checkbox"
          value="Comedy"
        />
        Comedy
      </label>
    </div>
    <div
      class="input-wrapper"
    >
      <label
        class="checkbox-input"
      >
        <input
          id="movieGenres"
          label="Action"
          name="movieGenres"
          type="checkbox"
          value="Action"
        />
        Action
      </label>
    </div>
    <div
      class="input-wrapper"
    >
      <label
        class="checkbox-input"
      >
        <input
          id="movieGenres"
          label="Drama"
          name="movieGenres"
          type="checkbox"
          value="Drama"
        />
        Drama
      </label>
    </div>
    <div
      class="input-wrapper"
    >
      <label
        class="checkbox-input"
      >
        <input
          id="movieGenres"
          label="Horror"
          name="movieGenres"
          type="checkbox"
          value="Horror"
        />
        Horror
      </label>
    </div>
    <div
      class="input-wrapper"
    >
      <label
        class="checkbox-input"
      >
        <input
          id="movieGenres"
          label="Sci-Fi"
          name="movieGenres"
          type="checkbox"
          value="Sci-Fi"
        />
        Sci-Fi
      </label>
    </div>
  </div>
</div>
`;
